### PR TITLE
Update BuildFromSource.md to explain CS0006 errors

### PR DESCRIPTION
**Pull request by natemcmaster**
Related to https://github.com/aspnet/AspNetCore/issues/7740

cc @aspnet/build @smitpatel

### CODE REVIEW

File: docs/BuildFromSource.md [hunk 1]
```diff
@@ -81,21 +81,24 @@ Instead, we have many .sln files which include a sub-set of projects. These prin
...
+### Common error: CS0006
 
-Opening solution files may produce an error code NU1105 with a message such
+Opening solution files and building may produce an error code CS0006 with a message such
 
-> Unable to find project information for 'C:\src\AspNetCore\src\Hosting\Abstractions\src\Microsoft.AspNetCore.Hosting.Abstractions.csproj'. Inside Visual Studio, this may be because the project is unloaded or not part of current solution. Otherwise the project file may be invalid or missing targets required for restore.
+> Error CS0006 Metadata file 'C:\src\aspnet\AspNetCore\artifacts\bin\Microsoft.AspNetCore.Metadata\Debug\netstandard2.0\Microsoft.AspNetCore.Metadata.dll' could not be found
 
-This is a known issue in NuGet (<https://github.com/NuGet/Home/issues/5820>) and we are working with them for a solution. See also <https://github.com/aspnet/AspNetCore/issues/4183> to track progress on this.
+The cause of this problem is that the solution you are using does not include the project that produces this .dll. This most often occurs after we have added new projects to the repo, but failed to update our .sln files to include the new project, but it is sometimes the indented behavior.
```
Comment (HaoK) [inline docs/BuildFromSource.md]: nit: intended (not indented?)

File: docs/BuildFromSource.md [hunk 2]
```diff
@@ -81,21 +81,24 @@ Instead, we have many .sln files which include a sub-set of projects. These prin
 
 > :bulb: Pro tip: `dotnet new sln` and `dotnet sln` are one of the easiest ways to create and modify solutions.
 
-### Known issue: NU1105
+### Common error: CS0006
```
Comment (dougbu) [inline docs/BuildFromSource.md]: Can external contributors who are not themselves adding a new project hit this? That is, doesn't `CodeCheck.ps1` prevent seeing this error in 'master'?

**Review by dougbu — APPROVED** [review]
Modulo the typo @HaoK caught

File: docs/BuildFromSource.md (hunk 2 above)
Comment (natemcmaster) [reply docs/BuildFromSource.md]: Yes. CodeCheck.ps1 only checks that each .csproj referenced from a .sln exists. It doesn't check that the entire project reference graph has been added to the solution.

**Pull request merged**